airbyte-webapp/src/hooks/services/useConnectionHook.tsx

```diff
@@ -230,12 +230,23 @@ const useUpdateConnection = () => {
  */
 export const useEnableConnection = () => {
   const analyticsService = useAnalyticsService();
+  const { trackError } = useAppMonitoringService();
+  const { registerNotification } = useNotificationService();
+  const { formatMessage } = useIntl();
   const { mutateAsync: updateConnection } = useUpdateConnection();
 
   return useMutation(
     ({ connectionId, enable }: { connectionId: WebBackendConnectionUpdate["connectionId"]; enable: boolean }) =>
       updateConnection({ connectionId, status: enable ? ConnectionStatus.active : ConnectionStatus.inactive }),
     {
+      onError: (error: Error) => {
+        trackError(error);
+        registerNotification({
+          id: `tables.updateFailed.${error.message}`,
+          text: `${formatMessage({ id: "connection.updateFailed" })}: ${error.message}`,
+          type: ToastType.ERROR,
+        });
+      },
```

Comment:
Displays an error toast if enabling/disabling the connection failed (e.g. due to credit balance)

https://user-images.githubusercontent.com/7550957/217348029-9af939eb-f6ba-41c7-931d-b6d959b7fb34.mov